<?xml version="1.0"?>

<!--
Mon Nov 21 17:30:20 EET 2016, Nikos Koukis

Setup the model for a single robot agent in the Gazebo simulation environment.
Launchfile *does not* set up the graphslam-related processes. Refer to
setup_single_graphslam_agent_simul.launch for the latter

-->

<launch>
    <arg name="output" value="$(env MR_OUTPUT_MESSAGES_TO)"/>
    <arg name="launch_dir" value="$(find csl_robots_gazebo)/launch" />
    <arg name="camera_name" value="camera1" />

    <arg name="robot_name" default="pioneer_3at" />
    <arg name="robot_type" default="pioneer_3at" />
    <arg name="position_args" default="-x 0.0 -y 0.0 -z 0.02" />


    <!-- Launch ar_sys nodes only in multi-robot graphSLAM setup -->
    <arg name="launch_ar_sys" value="$(env MR_IS_MULTIROBOT_GRAPHSLAM)" />

    <arg name="start_robot_control" value="$(env MR_USE_RQT_ROBOT_STEERING)"/>

    <!-- Upload the robot_description -->
    <param name="/robot_description"
        command="$(find xacro)/xacro --inorder $(find csl_robot_descriptions)/defs/$(arg robot_type).xacro camera_name:=$(arg camera_name)" />

    <!-- publish TFs for static links -->
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="$(arg output)">
        <param name="publish_frequency" type="double" value="30.0"/>
    </node>

    <!-- push robot_description to factory and spawn robot in gazebo -->
    <node name="spawn_pioneer" pkg="gazebo_ros" type="spawn_model"
        args="$(arg position_args) -urdf -param /robot_description -model $(arg robot_name) -namespace /$(arg robot_name)"
        respawn="false" output="$(arg output)" />

    <!-- Launch rsys -->
    <group if="$(arg launch_ar_sys)">
        <include file="$(find csl_robots_gazebo)/launch/setup_single_ar_sys.launch">
            <arg name="robot_name" value="$(arg robot_name)" />
            <arg name="camera_name" value="$(arg camera_name)" />
        </include>
    </group>

    <!-- Launch controls -->
    <!-- rqt is shit at exporting/importing perspectives - do everything manually! -->
    <group if="$(arg start_robot_control)">
        <include file="$(arg launch_dir)/setup_single_robot_control.launch">
            <arg name="robot_name" value="$(arg robot_name)"/>
        </include>
    </group>

</launch>
